api/src/main/java/org/openmrs/module/reporting/report/service/ReportService.java

```diff
@@ -127,19 +127,75 @@ public interface ReportService extends OpenmrsService {
+	 * @param requestOnOrAfter a Date used to limit result to ReportRequests which ware requested on or after
+	 *                          (greater or equal filter), nullable
+	 * @param requestOnOrBefore a Date used to limit result to ReportRequests which ware requested on or before
+	 *                          (lower or equal filter), nullable
+	 * @param statuses an array of Status, used to limit result to ReportRequests with status included in the array, null
+	 *                  or empty array means that all statuses are included, nullable
 	 * @return all {@link ReportRequest} in the system that match the passed parameters
 	 * @should retrieve report requests by definition
 	 */
 	@Transactional(readOnly = true)
 	public List<ReportRequest> getReportRequests(ReportDefinition reportDefinition, Date requestOnOrAfter, Date requestOnOrBefore, Status...statuses);
 
 	/**
+	 * Get Report Requests by Report Definition, request date, having given status, limit result to at most {@code
+	 * mostRecentNum} elements.
+	 * The list is sorted descending by evaluateCompleteDatetime, evaluateStartDatetime, priority, requestDate.
+	 *
+	 * @param reportDefinition a Report Definition filter, nullable
+	 * @param requestOnOrAfter a Date used to limit result to ReportRequests which ware requested on or after
+	 *                          (greater or equal filter), nullable
+	 * @param requestOnOrBefore a Date used to limit result to ReportRequests which ware requested on or before
+	 *                          (lower or equal filter), nullable
+	 * @param mostRecentNum maximum number of results, not null
+	 * @param statuses an array of Status, used to limit result to ReportRequests with status included in the array, null
+	 *                  or empty array means that all statuses are included, nullable
 	 * @return all {@link ReportRequest} in the system that match the passed parameters
 	 * @should retrieve report requests by definition
 	 */
 	@Transactional(readOnly = true)
 	public List<ReportRequest> getReportRequests(ReportDefinition reportDefinition, Date requestOnOrAfter, Date requestOnOrBefore, Integer mostRecentNum, Status...statuses);
 
+	/**
+	 * Get Report Requests by Report Definition, request date, having given status.
+	 * The list is sorted descending by evaluateCompleteDatetime, evaluateStartDatetime, priority, requestDate.
+	 *
+	 * @param reportDefinition a Report Definition filter, nullable
+	 * @param requestOnOrAfter a Date used to limit result to ReportRequests which ware requested on or after
+	 *                          (greater or equal filter), nullable
+	 * @param requestOnOrBefore a Date used to limit result to ReportRequests which ware requested on or before
+	 *                          (lower or equal filter), nullable
+	 * @param firstResult the first result to be retrieved, not null
+	 * @param maxResults a limit upon the number of Report Requests to be retrieved, not null

```

Comment:
The `maxResults` does appear that it can be nullable, based on the dao implementation...

Reply:
Done